src/index/concurrent_table.cpp

```diff
@@ -60,6 +60,15 @@ DataItem* ConcurrentTable::GetOrInsert(const std::string_view key) {
   return item;
 }
 
+bool ConcurrentTable::Insert(const std::string_view key) {
+  bool is_entry_null = index_->IsEntryNull(key);
+  if (is_entry_null) {
+    return index_->Insert(key);
+  }
+  bool is_entry_empty = index_->IsEntryEmpty(key);
+  return is_entry_empty;
```

Comment:
If `is_entry_empty == true`, check whether the key exists in `range_index`, and then insert?